packages/api/src/codegen/codegenerator.ts

```diff
@@ -28,8 +28,9 @@ export default abstract class CodeGenerator {
   requiredPackages!: Record<
     string,
     {
+      dependencyType: 'production' | 'development';
       reason: string;
-      url: string;
+      url?: string;
```

Comment:
Made `url` optional because it feels silly to write out https://typescriptlang.org/ as the URL for TS.